### PR TITLE
replace select asset type dialog by dropdown, move select language to top right

### PR DESCRIPTION
**Pull request by bugjt**
closes: #552 
Fix: 
- refactor dropdown component, change layout using op.defer
- change wallet/account modalSelector in send/receive page to dropdown
- change the back button arrow to the same style
- move the language dropdown to the top right of the setup page
- change the select asset dialog to dropdown on the setup page
- Fix layout of some dropdown
Screenshot 2024-07-02 at 04 45 10
Screenshot 2024-07-02 at 04 45 33
Screenshot 2024-07-02 at 04 45 52
Screenshot 2024-07-07 at 17 27 18
Screenshot 2024-07-07 at 17 27 26
Screenshot 2024-07-07 at 17 27 40
Screenshot 2024-07-07 at 17 27 48

### CODE REVIEW

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]
Update dropdown on some pages and modals bellow:

image

Send page and modal
image

image

Receive page and modal
image

Staking modal
image

**Comment by bugjt** [comment]
oke i got it

**Comment by dreacot** [comment]
the select asset type screenshot, should take the exact format as seen in the screenshot below

Screenshot 2024-07-02 at 3 14 02 PM

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]
Why do we need create new account and wallet dropdown on components?

The app crash when click to send button
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0xd0 pc=0x1035a4354]

goroutine 8 [running]:
github.com/crypto-power/cryptopower/ui/page/send.NewSendPage.func1()
        /Users/justindo/Works/cryptopower/ui/page/send/page.go:131 +0x24
github.com/crypto-power/cryptopower/ui/page/components.(*FeeRateSelector).ratesUnit(...)
        /Users/justindo/Works/cryptopower/ui/page/components/fee_rate_selector.go:251
github.com/crypto-power/cryptopower/ui/page/components.NewFeeRateSelector(0x140003db950, 0x1404b28be90)
        /Users/justindo/Works/cryptopower/ui/page/components/fee_rate_selector.go:89 +0x5b8
github.com/crypto-power/cryptopower/ui/page/send.NewSendPage(0x140003db950, {0x0, 0x0})
        /Users/justindo/Works/cryptopower/ui/page/send/page.go:133 +0x378
github.com/crypto-power/cryptopower/ui/page/root.(*HomePage).HandleUserInteractions(0x1400068ad00, {{{0x640, 0x514}, {0x640, 0x514}}, {0x40000000, 0x40000000}, {0xc19b00c8f7c18b58, 0x360d42668a, 0x105135540}, ...})
        /Users/justindo/Works/cryptopower/ui/page/root/home_page.go:396 +0x1318
github.com/crypto-power/cryptopower/ui.(*Window).handleFrameEvent(0x14000593260, {{0xc19b00c8f7c18b58, 0x360d42668a, 0x105135540}, {0x40000000, 0x40000000}, {0x640, 0x514}, {0x0, 0x0, ...}, ...})
        /Users/justindo/Works/cryptopower/ui/window.go:252 +0x1dc
github.com/crypto-power/cryptopower/ui.(*Window).HandleEvents(0x14000593260)
        /Users/justindo/Works/cryptopower/ui/window.go:222 +0x26c
main.main.func4()
        /Users/justindo/Works/cryptopower/main.go:121 +0x20
created by main.main in goroutine 1
        /Users/justindo/Works/cryptopower/main.go:119 +0x3a4
exit status 2
```

The received crash too
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x40 pc=0x101d65d24]

goroutine 15 [running]:
github.com/crypto-power/cryptopower/ui/page/components.(*WalletDropdown).ListenForTxNotifications(0x14000b48230, {0x102b6add0, 0x14000901770})
        /Users/justindo/Works/cryptopower/ui/page/components/wallet_dropdown.go:230 +0x114
github.com/crypto-power/cryptopower/ui/page/receive.(*Page).OnNavigatedTo(0x14001300588)
        /Users/justindo/Works/cryptopower/ui/page/receive/receive_page.go:156 +0x60
github.com/crypto-power/cryptopower/app.(*PageStack).PushAndNavigate(0x1400835f3e0, {0x102b651a8, 0x14001300588}, {0x102b68040, 0x1400d0c6890})
        /Users/justindo/Works/cryptopower/app/pagestack.go:67 +0x338
github.com/crypto-power/cryptopower/app.(*MasterPage).Display(0x1400d0c6890, {0x102b651a8?, 0x14001300588?})
        /Users/justindo/Works/cryptopower/app/masterpage.go:46 +0x3c
github.com/crypto-power/cryptopower/ui/page/wallet.(*SingleWalletMasterPage).HandleUserInteractions.func1({0x102b651a8, 0x14001300588})
        /Users/justindo/Works/cryptopower/ui/page/wallet/single_wallet_main_page.go:301 +0x44
github.com/crypto-power/cryptopower/ui/page/wallet.(*SingleWalletMasterPage).HandleUserInteractions(0x14000c438c8, {{{0x640, 0x514}, {0x640, 0x514}}, {0x40000000, 0x40000000}, {0xc19b00e940b79fd0, 0x68900189c, 0x103ab9540}, ...})
        /Users/justindo/Works/cryptopower/ui/page/wallet/single_wallet_main_page.go:341 +0x404
github.com/crypto-power/cryptopower/ui/page/root.(*HomePage).HandleUserInteractions(0x14000933380, {{{0x640, 0x514}, {0x640, 0x514}}, {0x40000000, 0x40000000}, {0xc19b00e940b79fd0, 0x68900189c, 0x103ab9540}, ...})
        /Users/justindo/Works/cryptopower/ui/page/root/home_page.go:361 +0xf8
github.com/crypto-power/cryptopower/ui.(*Window).handleFrameEvent(0x1400039af60, {{0xc19b00e940b79fd0, 0x68900189c, 0x103ab9540}, {0x40000000, 0x40000000}, {0x640, 0x514}, {0x0, 0x0, ...}, ...})
        /Users/justindo/Works/cryptopower/ui/window.go:252 +0x1dc
github.com/crypto-power/cryptopower/ui.(*Window).HandleEvents(0x1400039af60)
        /Users/justindo/Works/cryptopower/ui/window.go:222 +0x26c
main.main.func4()
        /Users/justindo/Works/cryptopower/main.go:121 +0x20
created by main.main in goroutine 1
        /Users/justindo/Works/cryptopower/main.go:119 +0x3a4
exit status 2
```

File: ui/page/components/wallet_setup_page.go [hunk 1]
```diff
@@ -491,7 +528,7 @@ func (pg *CreateWallet) createWallet() {
 	walletName := pg.walletName.Editor.Text()
 	pass := pg.passwordEditor.Editor.Text()
 	seedType := GetWordSeedType(pg.seedTypeDropdown.Selected())
-	switch *pg.assetTypeSelector.SelectedAssetType() {
+	switch getAssetType(pg.assetTypeDropdown.Selected()) {
```
Comment (dmigwi) [inline ui/page/components/wallet_setup_page.go:531]: This function call is not necessary, make `pg.assetTypeDropdown.Selected()` return an object of type `libutils.AssetType`.

File: ui/page/receive/receive_page.go [hunk 2]
```diff
@@ -42,16 +41,18 @@ type Page struct {
...
+	pageContainer   layout.List
+	scrollContainer *widget.List
+	isNewAddr       bool
+	currentAddress  string
+	qrImage         *image.Image
+	newAddr, copy   *cryptomaterial.Clickable
+	info            cryptomaterial.IconButton
+	card            cryptomaterial.Card
+
+	walletDropdown     *components.WalletDropdown
```
Comment (dmigwi) [inline ui/page/receive/receive_page.go:53]: why did you remove `components.WalletAndAccountSelector`?

**Review by dmigwi — CHANGES_REQUESTED** [review]
Here is my first review of the PR.

File: ui/page/receive/receive_page.go (hunk 2 above)
Comment (bugjt) [reply ui/page/receive/receive_page.go:53]: It has already been replaced by walletDropdown and accountDropdown

File: ui/page/components/account_dropdown.go [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+				item := cryptomaterial.DropDownItem{
+					Text:      fmt.Sprint(account.Number),
+					Icon:      d.Theme.Icons.AccountIcon,
+					DisplayFn: d.getAccountItemLayout(account),
+				}
+				items = append(items, item)
+				d.allAccounts = append(d.allAccounts, account)
+			}
+		}
+		if len(items) > 0 && (d.selectedAccount == nil || !d.selectedIsValid(d.selectedAccount) || !d.accountIsValid(d.selectedAccount)) {
```
[the remark this answers is not in the document]
Comment (bugjt) [reply ui/page/components/account_dropdown.go:65]: if not set, all account is invalid

File: ui/page/components/account_dropdown.go [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+	selectedAccount        *sharedW.Account
+	selectedWallet         sharedW.Asset
+	dropdown               *cryptomaterial.DropDown
+	allAccounts            []*sharedW.Account
+	accountChangedCallback func(*sharedW.Account)
+	accountIsValid         func(*sharedW.Account) bool
+	isWatchOnlyEnabled     bool
+}
+
+func NewAccountDropdown(l *load.Load, wallet sharedW.Asset) *AccountDropdown {
```
[the remark this answers is not in the document]
Comment (bugjt) [reply ui/page/components/account_dropdown.go:27]: I separated it into two distinct components to make tracking and debugging easier

File: ui/page/components/wallet_setup_page.go [hunk 5]
```diff
@@ -262,74 +297,76 @@ func (pg *CreateWallet) walletTypeSection(gtx C) D {
...
-				}
+					if pg.selectedWalletAction == i {
+						col = pg.Theme.Color.White
+						title.Color = pg.Theme.Color.Primary
 
-				if item.clickable.IsHovered() {
-					item.border.Color = pg.Theme.Color.Gray1
-					title.Color = pg.Theme.Color.Gray1
-				}
+						if pg.AssetsManager.IsDarkModeOn() {
```
[the remark this answers is not in the document]
Comment (bugjt) [reply ui/page/components/wallet_setup_page.go]: okay, I deleted it

File: ui/page/components/wallet_setup_page.go [hunk 6]
```diff
@@ -135,6 +135,34 @@ func NewCreateWallet(l *load.Load, walletCreationSuccessCallback func(), assetTy
...
+			Icon: l.Theme.AssetIcon(assType),
+		}
+		items = append(items, item)
+	}
+
+	assetTypeDropdown := l.Theme.NewCommonDropDown(items, nil, values.MarginPadding340, values.AssetTypeDropdownGroup, false)
+	return assetTypeDropdown
+}
+
+func getAssetType(assetTypeStr string) libutils.AssetType {
```
[the remark this answers is not in the document]
Comment (bugjt) [reply ui/page/components/wallet_setup_page.go]: I deleted it

File: ui/page/components/wallet_dropdown.go [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+	selectedWallet        sharedW.Asset
+	dropdown              *cryptomaterial.DropDown
+	allWallets            []sharedW.Asset
+	walletChangedCallback func(sharedW.Asset)
+	walletIsValid         func(sharedW.Asset) bool
+	isWatchOnlyEnabled    bool
+	assetTypes            []utils.AssetType
+}
+
+func NewWalletDropdown(l *load.Load, assetType ...utils.AssetType) *WalletDropdown {
```
[the remark this answers is not in the document]
Comment (bugjt) [reply ui/page/components/wallet_dropdown.go:28]: same above comment

**Comment by bugjt** [comment]
I fixed it and refactored according to your comments

**Comment by dmigwi** [comment]
>I separated it into two distinct components to make tracking and debugging easier

@bugjt Separating the implementation into two distinct components is a really nice idea.  Although a new problem (code duplication) has been introduced. If someone makes an improvement on `func NewWalletAndAccountSelector(`, the new account and wallet implementations won't benefit from it.

Here are the ways we could make progress going forward:
1. Rename `wallet_dropdown.go` => `wallet_selector.go` and `account_dropdown.go` => `account_selector.go`. **Reasoning: ->** *Your implementation in those files mostly covers wallet/account selectors (more than 98%).*

2. Move the dropdown implementation to the individual files they are required. This shouldn't introduce much code duplication.
3. Create a selector interface that both account and wallet selector must implement. In the process, ensure common implementation between the two interface implementations is shared. Avoid code duplication where possible. 
**Reasoning: ->** *It will become super easy to swap either of implementations without adding any extra code.*
4. Finally replace `func NewWalletAndAccountSelector(` with either of the two interfaces implemented depending on the exact use case. 
**Reasoning:->** *Points 1-3 have rendered `NewWalletAndAccountSelector` implementation obsolete thus not necessary any more.*

By implementing the above issues, you will have ensured that your changes won't be erased by someone else in the near future during their optimisation effort too.

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]
- I still get crash when click to Receive button on main page

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x160 pc=0x103d3a814]

goroutine 12 [running]:
github.com/crypto-power/cryptopower/ui/page/receive.(*Page).OnNavigatedTo(0x14002782c08)
        /Users/justindo/Works/cryptopower/ui/page/receive/receive_page.go:151 +0x24
github.com/crypto-power/cryptopower/ui/page/receive.(*Page).OnResume(0x41800000?)
        /Users/justindo/Works/cryptopower/ui/page/receive/receive_page.go:143 +0x1c
github.com/crypto-power/cryptopower/app.(*SimpleWindowNavigator).ShowModal(0x14000c576b0, {0x10498adc0, 0x14002782c08})
        /Users/justindo/Works/cryptopower/app/window.go:105 +0x1d0
github.com/crypto-power/cryptopower/ui/page/root.(*HomePage).HandleUserInteractions(0x1400011b860, {{{0x640, 0x514}, {0x640, 0x514}}, {0x40000000, 0x40000000}, {0xc19c46eab6a39838, 0x82f4008aa, 0x1058d54a0}, ...})
        /Users/justindo/Works/cryptopower/ui/page/root/home_page.go:386 +0x5ec
github.com/crypto-power/cryptopower/ui.(*Window).handleFrameEvent(0x1400091c8a0, {{0xc19c46eab6a39838, 0x82f4008aa, 0x1058d54a0}, {0x40000000, 0x40000000}, {0x640, 0x514}, {0x0, 0x0, ...}, ...})
        /Users/justindo/Works/cryptopower/ui/window.go:252 +0x1dc
github.com/crypto-power/cryptopower/ui.(*Window).HandleEvents(0x1400091c8a0)
        /Users/justindo/Works/cryptopower/ui/window.go:222 +0x26c
main.main.func4()
        /Users/justindo/Works/cryptopower/main.go:121 +0x20
created by main.main in goroutine 1
        /Users/justindo/Works/cryptopower/main.go:119 +0x3a4
exit status 2
```

Please check logic when you click send button on home page, I can't select the wallet on them

image

- My d2 wallet have 3 accounts but I only see 2 account on select account in send page

image

- I can't close dropdown when click outside dropdown, please check it again
- Staking page too
- Create wallet same
image
image


- Dropdown on restore wallet is too long
image

image

- Hex tab the same
image

File: ui/page/root/home_page.go [hunk 8]
```diff
@@ -131,15 +131,15 @@ func (hp *HomePage) initPageItems() {
...
 	// Add trade tab only if not on macOS
-	if !appos.Current().IsDarwin() {
-		// Determine the insertion point, which is second to last position
-		insertionPoint := len(navigationTabTitles) - 1
-		if insertionPoint < 0 {
-			insertionPoint = 0
-		}
-		// Append at the second to last position
-		navigationTabTitles = append(navigationTabTitles[:insertionPoint], append([]string{values.String(values.StrTrade)}, navigationTabTitles[insertionPoint:]...)...)
+	// if !appos.Current().IsDarwin() {
```
Comment (dreacot) [inline ui/page/root/home_page.go:134]: why is this commented out?

File: ui/page/root/overview_page.go [hunk 9]
```diff
@@ -355,17 +354,17 @@ func (pg *OverviewPage) layoutMobile(gtx C) D {
...
-	if !appos.Current().IsIOS() || !appos.Current().IsDarwin() {
-		// Determine the insertion point, which is second to last position
-		insertionPoint := len(pageContent) - 1
-		if insertionPoint < 0 {
-			insertionPoint = 0
-		}
-
-		// Append at the second to last position
-		pageContent = append(pageContent[:insertionPoint], append([]func(gtx C) D{pg.recentTrades}, pageContent[insertionPoint:]...)...)
+	// if !appos.Current().IsIOS() || !appos.Current().IsDarwin() {
```
Comment (dreacot) [inline ui/page/root/overview_page.go:357]: why is this commented out?

**Review by dreacot — CHANGES_REQUESTED** [review]

File: ui/page/root/home_page.go (hunk 8 above)
Comment (bugjt) [reply ui/page/root/home_page.go:134]: sorry, I forgot revert this code when testing this pull request, I will uncomment it.

File: ui/page/root/overview_page.go (hunk 9 above)
Comment (bugjt) [reply ui/page/root/overview_page.go:357]: same above

File: ui/page/components/wallet_dropdown.go [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+}
+
+func (d *WalletDropdown) SetupWallet() {
+	d.allWallets = make([]sharedW.Asset, 0)
+	wallets := d.AssetsManager.AssetWallets(d.assetTypes...)
+	items := []cryptomaterial.DropDownItem{}
+	if len(wallets) > 0 {
+		for _, w := range wallets {
+			if w.IsWatchingOnlyWallet() && !d.isWatchOnlyEnabled || d.walletIsValid != nil && !d.walletIsValid(w) {
+				return
```
Comment (JustinBeBoy) [inline ui/page/components/wallet_dropdown.go:47]:
```suggestion
				continue
```

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]

**Comment by JustinBeBoy** [comment]
@bugjt thanks for your efforts.

Seem like close dropdown on send page/modal not works, I have research and debug the reason cause that bug, don't use `handle()` in `recipientLayout()`, please move it to `HandleUserInteractions()` in send page

**Comment by bugjt** [comment]
thank you, I have fixed it

File: ui/page/components/account_dropdown.go [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+}
+
+func (d *AccountDropdown) SetAccountItems(wallet sharedW.Asset) {
+	if wallet == nil {
+		return
+	}
+	d.selectedWallet = wallet
+	items := []cryptomaterial.DropDownItem{}
+	d.allAccounts = make([]*sharedW.Account, 0)
+	if wallet != nil && (!wallet.IsWatchingOnlyWallet() || d.isWatchOnlyEnabled) {
```
Comment (JustinBeBoy) [inline ui/page/components/account_dropdown.go:45]:
```suggestion
	if !wallet.IsWatchingOnlyWallet() || d.isWatchOnlyEnabled {
```

File: ui/page/components/wallet_dropdown.go [hunk 12]
```diff
@@ -0,0 +1,264 @@
...
+	d.isWatchOnlyEnabled = isEnable
+	return d
+}
+
+func (d *WalletDropdown) SetSelectedWallet(wallet sharedW.Asset) {
+	if wallet == nil {
+		return
+	}
+	d.dropdown.SetSelectedValue(fmt.Sprint(wallet.GetWalletID()))
+	// d.onChanged()
```
Comment (JustinBeBoy) [inline ui/page/components/wallet_dropdown.go:106]: remove comment

File: ui/page/components/wallet_dropdown.go [hunk 13]
```diff
@@ -0,0 +1,264 @@
...
+	for _, wallet := range d.allWallets {
+		if wallet.GetWalletID() == walletID {
+			return wallet
+		}
+	}
+	return nil
+}
+
+func (d *WalletDropdown) onChanged() {
+	if d.dropdown != nil {
```
Comment (JustinBeBoy) [inline ui/page/components/wallet_dropdown.go:175]: I don't see any case of dropdown nil

File: ui/page/components/wallet_dropdown.go [hunk 14]
```diff
@@ -0,0 +1,264 @@
...
+func (d *WalletDropdown) Handle(gtx C) {
+	if d.dropdown.Changed(gtx) {
+		d.onChanged()
+	}
+}
+
+func (d *WalletDropdown) Build() *WalletDropdown {
+	d.SetupWallet()
+	return d
+}
```
Comment (JustinBeBoy) [inline ui/page/components/wallet_dropdown.go:207]: why you don't return `*WalletDropdown` from `SetupWallet()` and remove this function ?

File: ui/page/components/wallet_dropdown.go [hunk 15]
```diff
@@ -0,0 +1,264 @@
...
+	return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			lbl := d.Theme.H6(values.String(titleKey))
+			lbl.TextSize = values.TextSizeTransform(d.IsMobileView(), values.TextSize16)
+			lbl.Font.Weight = font.SemiBold
+			return layout.Inset{Bottom: values.MarginPadding4}.Layout(gtx, lbl.Layout)
+		}),
+		layout.Rigid(func(gtx C) D {
+			return d.dropdown.Layout(gtx)
+		}),
```
Comment (JustinBeBoy) [inline ui/page/components/wallet_dropdown.go:219]:
```suggestion
		layout.Rigid(d.dropdown.Layout),
```

File: ui/page/components/wallet_setup_page.go [hunk 16]
```diff
@@ -227,9 +234,17 @@ func (pg *CreateWallet) Layout(gtx C) D {
...
+									layout.Expanded(func(gtx layout.Context) layout.Dimensions {
+										return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+											layout.Rigid(pg.walletOptions),
+										)
+									}),
+									layout.Expanded(func(gtx layout.Context) layout.Dimensions {
+										return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+											layout.Rigid(pg.walletTypeSection),
+										)
+									}),
```
Comment (JustinBeBoy) [inline ui/page/components/wallet_setup_page.go:247]: I don't see any reason use `layout.Flex` in here

```suggestion
									layout.Expanded(pg.walletOptions),
									layout.Expanded(pg.walletTypeSection),
```

File: ui/page/dcrdex/dex_onboarding_page.go [hunk 17]
```diff
@@ -587,26 +587,25 @@ func (pg *DEXOnboarding) stepPostBond(gtx C) D {
 		layout.Rigid(func(gtx C) D {
 			return centerLayout(gtx, 0, 0, pg.Theme.Body1(values.String(values.StrSelectBondWalletMsg)).Layout)
 		}),
+		// layout.Rigid(func(gtx C) D {
```
Comment (JustinBeBoy) [inline ui/page/dcrdex/dex_onboarding_page.go:590]: remove comment if not need

File: ui/page/staking/stake_modal.go [hunk 18]
```diff
@@ -29,8 +29,10 @@ type ticketBuyerModal struct {
 
 	balToMaintainEditor cryptomaterial.Editor
 
-	accountSelector *components.WalletAndAccountSelector
-	vspSelector     *components.VSPSelector
+	// accountSelector *components.WalletAndAccountSelector
```
Comment (JustinBeBoy) [inline ui/page/staking/stake_modal.go:32]: remove if not need

File: ui/page/staking/stake_modal.go [hunk 19]
```diff
@@ -74,8 +76,8 @@ func (tb *ticketBuyerModal) OnResume() {
 		return
 	}
 
-	tb.initializeAccountSelector()
-	tb.accountSelector.ListenForTxNotifications(tb.ParentWindow()) // listener is stopped in OnDismissed()
+	tb.initializeAccountSelector(tb.dcrImpl)
+	// tb.accountSelector.ListenForTxNotifications(tb.ParentWindow()) // listener is stopped in OnDismissed()
```
Comment (JustinBeBoy) [inline ui/page/staking/stake_modal.go:80]: remove comment

File: ui/page/staking/stake_modal.go [hunk 20]
```diff
@@ -193,15 +193,16 @@ func (tb *ticketBuyerModal) initializeAccountSelector() {
...
 			return accountIsValid
-		})
-	_ = tb.accountSelector.SelectFirstValidAccount(tb.dcrImpl)
+		}).Build()
+	_ = tb.accountDropdown.SelectFirstValidAccount(tb.dcrImpl)
 }
 
 func (tb *ticketBuyerModal) OnDismiss() {
-	tb.accountSelector.StopTxNtfnListener()
+	// tb.accountSelector.StopTxNtfnListener()
```
Comment (JustinBeBoy) [inline ui/page/staking/stake_modal.go:201]: remove comment

File: ui/window.go [hunk 21]
```diff
@@ -290,7 +290,9 @@ func (win *Window) prepareToDisplayUI(gtx layout.Context) {
 		if modal == nil {
 			return layout.Dimensions{}
 		}
-		return modal.Layout(gtx)
+		return win.load.Theme.DropdownBackdrop.Layout(gtx, func(gtx C) D {
+			return modal.Layout(gtx)
+		})
```
Comment (JustinBeBoy) [inline ui/window.go:295]:
```suggestion
		return win.load.Theme.DropdownBackdrop.Layout(gtx, modal.Layout)
```

File: ui/window.go [hunk 22]
```diff
@@ -290,7 +290,9 @@ func (win *Window) prepareToDisplayUI(gtx layout.Context) {
 		if modal == nil {
 			return layout.Dimensions{}
```
Comment (JustinBeBoy) [inline ui/window.go:291]:
```suggestion
			return D{}
```

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]
Please clean codes

**Comment by bugjt** [comment]
It's ready for review

**Review by dmigwi — APPROVED** [review]
LGTM

**Review by dreacot — APPROVED** [review]
